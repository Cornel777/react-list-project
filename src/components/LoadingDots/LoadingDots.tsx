import { FunctionComponent } from "react";
import LoadingDotsContainer from "./styled/LoadingDotsContainer"

interface LoadingDotsProps {
  size?: 'small' | 'medium' | 'large';
  dataTestId?: string;
}

const LoadingDots: FunctionComponent<LoadingDotsProps> = ({dataTestId, size = 'large'}) => {
  return <LoadingDotsContainer size={size} dataTestId={dataTestId}>
    <span>.</span>
    <span>.</span>
    <span>.</span>
  </LoadingDotsContainer>;
};

export default LoadingDots;