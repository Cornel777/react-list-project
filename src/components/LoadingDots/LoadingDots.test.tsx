import { render } from '@testing-library/react';
import { describe, it, expect } from 'vitest';
import '@testing-library/jest-dom';
import LoadingDots from './LoadingDots';

describe('LoadingDots', () => {
  const renderLoadingDots = (props = {}) => {
    return render(
        <LoadingDots {...props} />
    );
  };

  it('renders loading dots', () => {
    const { container } = renderLoadingDots();
    const dots = container.querySelectorAll('span');
    expect(dots).toHaveLength(3);
  });

  it('renders with default large size', () => {
    const { container } = renderLoadingDots();
    expect(container.firstChild).toHaveStyle('font-size: 3rem');
  });

  it('renders with small size when specified', () => {
    const { container } = renderLoadingDots({ size: 'small' });
    expect(container.firstChild).toHaveStyle('font-size: 1rem');
  });

  it('matches snapshot', () => {
    const { container } = renderLoadingDots();
    expect(container).toMatchSnapshot();
  });
});