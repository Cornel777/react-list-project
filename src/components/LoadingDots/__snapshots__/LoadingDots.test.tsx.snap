// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`LoadingDots > matches snapshot 1`] = `
<div>
  <div
    class="sc-KXCwU jZTKhe"
    data-testid="loading-dots"
  >
    <span>
      .
    </span>
    <span>
      .
    </span>
    <span>
      .
    </span>
  </div>
</div>
`;
